packages/block-library/src/video/video-player.ios.js

```diff
@@ -0,0 +1,73 @@
+/**
+ * WordPress dependencies
+ */
+import { Component } from '@wordpress/element';
+import { Dashicon } from '@wordpress/components';
+
+/**
+ * External dependencies
+ */
+import { View, TouchableOpacity } from 'react-native';
+import { default as VideoPlayer } from 'react-native-video';
+
+/**
+ * Internal dependencies
+ */
+import styles from './video-player.scss';
+
+class Video extends Component {
+	constructor() {
+		super( ...arguments );
+		this.state = {
+			isLoaded: false,
+		};
+		this.onPressPlay = this.onPressPlay.bind( this );
+		this.onLoad = this.onLoad.bind( this );
+		this.onLoadStart = this.onLoadStart.bind( this );
+	}
+
+	onLoad() {
+		this.setState( { isLoaded: true } );
+	}
+
+	onLoadStart() {
+		this.setState( { isLoaded: false } );
+	}
+
+	onPressPlay() {
+		if ( this.player ) {
+			this.player.presentFullscreenPlayer();
+		}
+	}
+
+	render() {
+		const { style } = this.props;
+		const { isLoaded } = this.state;
+
+		return (
+			<View style={ styles.videoContainer }>
+				<VideoPlayer
+					{ ...this.props }
+					ref={ ( ref ) => {
+						this.player = ref;
+					} }
+					// Using built-in player controls is messing up the layout on iOS.
+					// So we are setting controls=false and adding a play button that
+					// will trigger presentFullscreenPlayer()
```

Comment:
Sounds good as a workaround for this PR, but let's make sure to track this in another issue for 2.0.

Reply:
I've added a new item to the TODO list here: https://github.com/wordpress-mobile/gutenberg-mobile/issues/687
Unfortunately this is in library level, they are setting the frame (as a workaround for some issue) when controls=true
https://github.com/react-native-community/react-native-video/blob/c30f246f5694452f4f081ff30ac4523dfd89f41d/ios/Video/RCTVideo.m#L593
And by doing that our they are messing up the y position of our page. I couldn't come up with a nice solution myself, I was planning to open an issue to their repo after this is merged. I think they might agree that current way is not very healthy. I can also try to tackle this more and send a PR to the original repo if they ignore the issue in the long run 🤔